helixbass/tsc-rust
# Parse + bind if-statement

In this PR:
- parse + bind an `if`/`else` statement

To test:
If you run `cargo run tmp.tsx` against a source file containing eg:
```
if (true) {
    const a = 1;
} else {
    const b = 2;
}

++a
```
then you should see `post-binding:` debug-logging where the `SourceFile`'s first `Statement` is an `IfStatement` whose `expression` is a `TrueKeyword` and whose `then_statement` and `else_statement` are both `Block`'s whose `statements` contain a single `VariableStatement`, and `parent` bindings should be initialized in them. Then it should panic during type-checking

Based on `fix-object-literal-checking`
